src/qibo/backends/__init__.py

```diff
@@ -140,8 +141,13 @@ def _default_transpiler(cls):  # pragma: no cover
         qubits = cls._backend.qubits
         natives = cls._backend.natives
         connectivity_edges = cls._backend.connectivity
-        if qubits is not None and natives is not None and connectivity is not None:
-            natives_enum = NativeGates.from_gatelist(natives)
+        if (
+            qubits is not None
+            and natives is not None
+            and connectivity_edges is not None
+        ):
+            # natives_enum = NativeGates.from_gatelist(natives)
+            natives_enum = reduce(or_, [NativeGates[nat] for nat in natives])
```

Comment:
Another option could be to extend the `NativeGates.__getitem__` (the square brackets), in order to also accept a list of strings (instead of a single `str`), and implement essentially in this way.

I.e. something like:
```py
def __getitem__(self, key):
    if isinstance(key, str):
        return super().__getitem__(key)
    return reduce(or_, [type(self)[nat] for nat in natives])
```

In this way, you could move this implementation as well (together with the imports) in the `NativeGates` class, which is more scoped, simplifying this more general one, in which you could just invoke as:
```suggestion
            natives_enum = NativeGates[natives]
```
and it will be also available for usage in other places (if needed), as your former implementation.

Reply:
To override `__getitem__`, I need to use `EnumMeta`. Is it okay to switch from `Flag` to `EnumMeta`?

https://stackoverflow.com/questions/24716723/issue-extending-enum-and-redefining-getitem

https://stackoverflow.com/questions/60374784/python-enum-from-string-raise-error-if-string-not-in-members

Reply:
Hi @csookim, thanks for digging into this, I did not realize the issue, but that's correct, because we're talking about the `__getitem__` of the class itself, not of its instances (that would be `NativeGates(...)[key]`, instead of `NativeGates[key]`).

#### Context

Maybe you're already aware, but to give you some context, even classes in Python are objects, and they are instances of other classes. When the instances of a class are still classes, that class is called a metaclass (but other than that, is a normal Python class).
The root class for all classes that are not metaclasses is [`object`](https://docs.python.org/3/library/functions.html#object), while the root class for metaclasses is [`type`](https://docs.python.org/3/library/functions.html#type) (in case you didn't know... surprise! `type()` is not a function, but a class constructor - but it can *also* be used essentially as a function, in the usual way).

#### Proposal

You could see that `Flag` and `EnumMeta` are not alternative to each other, e.g. in an answer in the first question you posted:
https://stackoverflow.com/a/24717640/8653979

The metaclass, set with the `metaclass=` keyword-like arguments to the class definition (aside inheritance), is not affecting the instances of the class you're defining, but the class itself as an object. That, as clarified above, it's exactly what we need to change.

So, what you can do is:
```py
from enum import Flag, EnumMeta

class FlagListAccess(EnumMeta):
    def __getitem__(self, key):
        # here the implementation above

class NativeGates(Flag, metaclass=FlagListAccess):
    ...
```
so it can keep being a `Flag`, you don't need to switch. And we don't need to change the behavior of the `NativeGates` instances - it is just an addition that will affect the access, i.e. only what we need.

---

Btw, notice that now `EnumMeta` has been renamed [`EnumType`](https://docs.python.org/3/library/enum.html#enum.EnumType).
However, this happened in py3.11, and we still support py3.9 and py3.10. Moreover, `EnumMeta` has been maintained as an alias for compatibility - so please, use `EnumMeta` for the time being (when we'll drop compatibility for py<3.11 we will switch).

Reply:
e913307

Reply:
I am not sure why, but using `reduce` with `super().__getitem__(key)` causes errors.

https://github.com/qiboteam/qibo/actions/runs/11361474890/job/31605243505?pr=1440#step:9:266

 I replaced it with a longer version.

08107cd